### ISSUE TITLE
sccache fails compiling target_build_utils crate

### ISSUE THREAD

**Issue by luser** [mozilla/sccache#91]
Reported by @Dushistov  in https://github.com/mozilla/sccache/issues/42#issuecomment-287605378, I can reproduce this locally:
```
luser@eye7:/build$ git clone https://github.com/nagisa/target_build_utils.rs
Cloning into 'target_build_utils.rs'...
remote: Counting objects: 90, done.
remote: Total 90 (delta 0), reused 0 (delta 0), pack-reused 90
Unpacking objects: 100% (90/90), done.
Checking connectivity... done.
luser@eye7:/build$ cd target_build_utils.rs/
luser@eye7:/build/target_build_utils.rs$ RUSTC=/build/sccache2/target/debug/rustc cargo build -v
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading phf v0.7.21
 Downloading serde_json v0.9.9
 Downloading phf_shared v0.7.21
 Downloading siphasher v0.2.2
 Downloading serde v0.9.11
 Downloading num-traits v0.1.37
 Downloading phf_codegen v0.7.21
 Downloading phf_generator v0.7.21
   Compiling siphasher v0.2.2
   Compiling num-traits v0.1.37
   Compiling libc v0.2.21
   Compiling itoa v0.3.1
   Compiling dtoa v0.4.1
   Compiling serde v0.9.11
     Running `/build/sccache2/target/debug/rustc --crate-name siphasher /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/siphasher-0.2.2/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=479c65b98dc022aa -C extra-filename=-479c65b98dc022aa --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --cap-lints allow`
     Running `/build/sccache2/target/debug/rustc --crate-name libc /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.21/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="use_std"' -C metadata=5dc7b85e748840b4 -C extra-filename=-5dc7b85e748840b4 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --cap-lints allow`
     Running `/build/sccache2/target/debug/rustc --crate-name itoa /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/itoa-0.3.1/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=7224f7b9cf4017b1 -C extra-filename=-7224f7b9cf4017b1 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --cap-lints allow`
     Running `/build/sccache2/target/debug/rustc --crate-name num_traits /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/num-traits-0.1.37/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=69f4129d46397118 -C extra-filename=-69f4129d46397118 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --cap-lints allow`
     Running `/build/sccache2/target/debug/rustc --crate-name serde /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-0.9.11/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg 'feature="std"' --cfg 'feature="default"' -C metadata=e9bfb02f34c97433 -C extra-filename=-e9bfb02f34c97433 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --cap-lints allow`
     Running `/build/sccache2/target/debug/rustc --crate-name dtoa /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/dtoa-0.4.1/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=14210f5fcc36f62c -C extra-filename=-14210f5fcc36f62c --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --cap-lints allow`
   Compiling phf_shared v0.7.21
     Running `/build/sccache2/target/debug/rustc --crate-name phf_shared /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/phf_shared-0.7.21/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=1f5276fa90b55cc9 -C extra-filename=-1f5276fa90b55cc9 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern siphasher=/build/target_build_utils.rs/target/debug/deps/libsiphasher-479c65b98dc022aa.rlib --cap-lints allow`
   Compiling rand v0.3.15
     Running `/build/sccache2/target/debug/rustc --crate-name rand /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.3.15/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=c9d9fbdab2355ee4 -C extra-filename=-c9d9fbdab2355ee4 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern libc=/build/target_build_utils.rs/target/debug/deps/liblibc-5dc7b85e748840b4.rlib --cap-lints allow`
   Compiling phf v0.7.21
     Running `/build/sccache2/target/debug/rustc --crate-name phf /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/phf-0.7.21/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=e3e74ffb62952a96 -C extra-filename=-e3e74ffb62952a96 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern phf_shared=/build/target_build_utils.rs/target/debug/deps/libphf_shared-1f5276fa90b55cc9.rlib --cap-lints allow`
   Compiling phf_generator v0.7.21
     Running `/build/sccache2/target/debug/rustc --crate-name phf_generator /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/phf_generator-0.7.21/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=6f65519ca5cfaed4 -C extra-filename=-6f65519ca5cfaed4 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern phf_shared=/build/target_build_utils.rs/target/debug/deps/libphf_shared-1f5276fa90b55cc9.rlib --extern rand=/build/target_build_utils.rs/target/debug/deps/librand-c9d9fbdab2355ee4.rlib --cap-lints allow`
   Compiling phf_codegen v0.7.21
     Running `/build/sccache2/target/debug/rustc --crate-name phf_codegen /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/phf_codegen-0.7.21/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=c32e3d0bb75511d3 -C extra-filename=-c32e3d0bb75511d3 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern phf_generator=/build/target_build_utils.rs/target/debug/deps/libphf_generator-6f65519ca5cfaed4.rlib --extern phf_shared=/build/target_build_utils.rs/target/debug/deps/libphf_shared-1f5276fa90b55cc9.rlib --cap-lints allow`
   Compiling target_build_utils v0.3.0 (file:///build/target_build_utils.rs)
     Running `/build/sccache2/target/debug/rustc --crate-name build_script_build build.rs --crate-type bin --emit=dep-info,link -C debuginfo=2 --cfg 'feature="serde_json"' --cfg 'feature="default"' -C metadata=f67e225c51abf503 -C extra-filename=-f67e225c51abf503 --out-dir /build/target_build_utils.rs/target/debug/build/target_build_utils-f67e225c51abf503 -L dependency=/build/target_build_utils.rs/target/debug/deps --extern phf_codegen=/build/target_build_utils.rs/target/debug/deps/libphf_codegen-c32e3d0bb75511d3.rlib`
     Running `/build/target_build_utils.rs/target/debug/build/target_build_utils-f67e225c51abf503/build-script-build`
   Compiling serde_json v0.9.9
     Running `/build/sccache2/target/debug/rustc --crate-name serde_json /home/luser/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-0.9.9/src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 -C metadata=523da6ef63875c40 -C extra-filename=-523da6ef63875c40 --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern serde=/build/target_build_utils.rs/target/debug/deps/libserde-e9bfb02f34c97433.rlib --extern itoa=/build/target_build_utils.rs/target/debug/deps/libitoa-7224f7b9cf4017b1.rlib --extern dtoa=/build/target_build_utils.rs/target/debug/deps/libdtoa-14210f5fcc36f62c.rlib --extern num_traits=/build/target_build_utils.rs/target/debug/deps/libnum_traits-69f4129d46397118.rlib --cap-lints allow`
     Running `/build/sccache2/target/debug/rustc --crate-name target_build_utils src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg 'feature="serde_json"' --cfg 'feature="default"' -C metadata=0c9859dfba606c3d -C extra-filename=-0c9859dfba606c3d --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern phf=/build/target_build_utils.rs/target/debug/deps/libphf-e3e74ffb62952a96.rlib --extern serde_json=/build/target_build_utils.rs/target/debug/deps/libserde_json-523da6ef63875c40.rlib`
error: Could not compile `target_build_utils`.

Caused by:
  process didn't exit successfully: `/build/sccache2/target/debug/rustc --crate-name target_build_utils src/lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --cfg feature="serde_json" --cfg feature="default" -C metadata=0c9859dfba606c3d -C extra-filename=-0c9859dfba606c3d --out-dir /build/target_build_utils.rs/target/debug/deps -L dependency=/build/target_build_utils.rs/target/debug/deps --extern phf=/build/target_build_utils.rs/target/debug/deps/libphf-e3e74ffb62952a96.rlib --extern serde_json=/build/target_build_utils.rs/target/debug/deps/libserde_json-523da6ef63875c40.rlib` (exit code: 254)
```

I'm using sccache built from rev 9a3447f4cc5b73069a4c046616f786992052c579 (current master) and Stable rust (1.16).

**Comment by luser** [comment]
Aha, running with server logs enabled shows:
```
DEBUG:sccache::server: [Error(Msg("Failed run rustc --dep-info. rustc stderr: `error: environment variable `OUT_DIR` not defined\n  --> src/lib.rs:98:18\n   |\n98 | include!(concat!(env!(\"OUT_DIR\"), \"/builtins.rs\"));\n   |                  ^^^^^^^^^^^^^^^\n\nerror: couldn\'t read \"src/0/builtins.rs\": No such file or directory (os error 2)\n  --> src/lib.rs:98:1\n   |\n98 | include!(concat!(env!(\"OUT_DIR\"), \"/builtins.rs\"));\n   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n`"), State { next_error: None })] compilation failed: "target_build_utils"
```

So this should be fixed by https://github.com/mozilla/sccache/issues/48 .

**Comment by luser** [comment]
I tested locally with my fixes for #48 and this crate compiles successfully now.
